src/map-view.android.ts

```diff
@@ -133,6 +134,45 @@ export class MapView extends MapViewBase {
                     onMyLocationButtonClick: () => {
                         owner.notifyMyLocationTapped();
 
+                        return false;
+                    },
+                }));
+
+                gMap.setOnIndoorStateChangeListener(new com.google.android.gms.maps.GoogleMap.OnIndoorStateChangeListener({
+                    onIndoorBuildingFocused: () => {
+                        const buildingFocused = gMap.getFocusedBuilding();
+                        const levels = [];
+                        let count = 0;
+                        while (count < buildingFocused.getLevels().length) {
+                            levels.push(
+                                {
+                                    name: buildingFocused.getLevels[count].getName(),
+                                    shortName: buildingFocused.getLevels[count].getShortName(),
+                                }
+                            );
+                            count += 1;
+                        }
+                        owner.notifyBuildingFocusedEvent(
+                            {
+                                defaultLevelIndex: buildingFocused.getDefaultLevelIndex(),
+                                levels: levels,
+                                isUnderground: buildingFocused.isUnderground(),
+                            }
+                        );
+
+                        return false;
+                    },
+                    onIndoorLevelActivated: (gmsIndoorBuilding) => {
+                        const level = gmsIndoorBuilding.getLevels()[gmsIndoorBuilding.getActiveLevelIndex()];
```

Comment:
.getLevels().get(gmsIndoorBuilding.getActiveLevelIndex());

Reply:
TypeError: buildingFocused.getLevels.get is not a function

Hehe, it should have been so easy but apparently not :p

Reply:
It was so simple at the end :p